src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs

```diff
@@ -12,6 +12,24 @@
 
 namespace MoBi.Engine.Sbml
 {
+   class CombinedUnit
+   {
+      public string Name { get => (_direct?.Name ?? "1") + (_inverse != null ? $"/{_inverse.Name}" : ""); }
+      public double Rate { get; private set; } = 1;
+      public void AddUnit(int kind, double exponent, double multiplier, IDictionary<int, Unit>  baseUnitsDictionary)
```

Comment:
Looks like a constructor to me no?

Reply:
Not really, if you check how it is used, you will see that it could be invoked twice. Once for setting the direct unit and a second time for setting the inverse unit (or in the other other, it does not really matter). Initially, I thought that unit definitions in an sbml file would only have one single unit but if you have "mole/l" for example you will have a unit definition with two units, one for mole with positive exponent and a second for liter with negative exponent. So we need to iterate first on all the units two collect all information (with AddUnit) to know if it is a plain inverse dimension (1/s for inverse time), a regular dimension (mole) or a combined dimension (mole/l).

Reply:
yep you are correct. Did not see the loop